### DISCUSSION TITLE
Understanding the use case for the typescript-resolvers plugin

### DISCUSSION THREAD

**Discussion by nimmolo** [dotansimha/graphql-code-generator#7131] [Q&A] [answered]
If I'm using resolvers for certain queries on my GraphQL server backend (graphql-ruby in my case) does that mean I should use the `typescript-resolvers` plugin in my codegen.yml config, to introspect those resolvers and their input types properly?

Or am I misunderstanding the use case, maybe it is for a different type of resolver? 

The [issue i've run into](https://github.com/ticruz38/graphql-codegen-svelte-apollo/issues/28) is i'm getting a malformed generated query type, where the type is explicitly omitting my query variables from a watched query, whereas i do want the variables in the query (because in my implementation, i want it to refetch when those variables change).

**Comment by alexrhogue** [comment, marked as answer]
I think the main reason (or at least my main reason) for using the `typescript-resolvers` plugin is to automatically generate types for an Apollo like resolver.  [Here is an example I made a few weeks ago](https://github.com/alexrhogue/odyssey-lift-off-typescript/blob/main/server/src/resolvers/index.ts). 

Without the plugin the typing of all your resolvers would be `(any, any, any, any)`.  This is not ideal in a TS project.  However, if you use the plugin and type your root resolver correctly you get strongly typed resolvers.  

```ts
 track?: Resolver<ResolversTypes['Track'], ParentType, ContextType, RequireFields<QueryTrackArgs, 'id'>>;
```

**Reply by nimmolo** [reply]
Thanks Alex. I understand a bit better... Now i'm pretty sure the issue with my generated query originates with either my code, or the typescript-svelte-apollo codegen plugin.
